pkg/api/api.go

```diff
@@ -117,6 +118,7 @@ func New(cfg Config, federationCfg tenantfederation.Config, serverCfg server.Con
 	if api.AuthMiddleware == nil {
 		api.AuthMiddleware = middleware.AuthenticateUser
 	}
+	api.AuthMiddleware = middleware.Merge(querierapi.ConsistencyMiddleware(), api.AuthMiddleware)
```

Comment:
It looks a bit weird wrapping the Auth middleware. This middleware is injected only on routes requiring authentication. Currently the consistency is just enforced on routes requiring auth, so it's not an issue, but conceptually it has nothing to do with auth (and doesn't require auth). I'm wondering if would be better just defining `ConsistencyMiddleware` on `API` and then always inject it in places where `AuthMiddleware` is injected.

Reply:
fair point. In 71ee8757dd5e760c9778ba68a1802c6603da6d86 i decided to not use indirection an directly add the `querierapi.ConsistencyMiddleware` to the handler. Let me know what you think